src/image.js

```diff
@@ -0,0 +1,121 @@
+require("babel-polyfill");
+
+import gm from "gm";
+import log from "./log";
+import uuid from "uuid";
+
+const im = gm.subClass({imageMagick: true});
+
+// Wrap these calls in promises so we can use async/await
+const promiseWrite = (client, file) => {
+  return new Promise((resolve, reject) => client.write(file, err => err ? reject(err) : resolve()));
+};
+const promiseSize = (client) => {
+  return new Promise((resolve, reject) => client.size((err, data) => err ? reject(err) : resolve(data)));
+};
+
+// Interlacing for png isn't efficient (both in filesize as render performance), so we only do it for jpg
+const interlace = async(client, params) => {
+  if (params.mime === 'image/jpeg') {
+    return client.interlace('Line');
+  }
+  return client;
+};
+
+// When cropping, we size the image first to fix completely within the bounding box
+// Then we crop the requested size, and center to the center of the image
+const crop = async(client, params) => {
+  // Resize the image to fit within the bounding box
+  // The ^ ensures the images is resized while maintaining the ratio
+  // However the sizes are treated as minimum values instead of maximum values
+  client = client.resize(params.width, params.height, '^');
+  const tmpFile = `/tmp/${uuid.v4()}`;
+  try {
+    await promiseWrite(client, tmpFile);
```

Comment:
Because imagemagick cannot perform this chain without re-reading the file. Which it can only do from disk.

File cleaning indeed should still happen